assets/js/modules/adsense/components/settings/AdBlockingRecoveryApp/SetupMain.test.js

```diff
@@ -0,0 +1,408 @@
+/**
+ * AdBlockingRecoveryWidget - SetupMain component test.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import { mockLocation } from '../../../../../../../tests/js/mock-browser-utils';
+import {
+	act,
+	fireEvent,
+	render,
+} from '../../../../../../../tests/js/test-utils';
+import {
+	createTestRegistry,
+	muteFetch,
+	provideModules,
+	provideSiteInfo,
+	unsubscribeFromAll,
+} from '../../../../../../../tests/js/utils';
+import { VIEW_CONTEXT_AD_BLOCKING_RECOVERY } from '../../../../../googlesitekit/constants';
+import { CORE_SITE } from '../../../../../googlesitekit/datastore/site/constants';
+import * as tracking from '../../../../../util/tracking';
+import { MODULES_ADSENSE } from '../../../datastore/constants';
+import SetupMain from './SetupMain';
+
+const mockTrackEvent = jest.spyOn( tracking, 'trackEvent' );
+mockTrackEvent.mockImplementation( () => Promise.resolve() );
+
+describe( 'AdBlockingRecoveryWidget - SetupMain', () => {
+	mockLocation();
+
+	let registry;
+	let container;
+	let getByRole;
+
+	beforeEach( () => {
+		mockTrackEvent.mockClear();
+		registry = createTestRegistry();
+		provideSiteInfo( registry );
+		provideModules( registry, [
+			{
+				active: true,
+				connected: true,
+				slug: 'adsense',
+			},
+		] );
+		registry.dispatch( MODULES_ADSENSE ).setSettings( {
+			adBlockingRecoverySetupStatus: '',
+		} );
+
+		const renderResult = render( <SetupMain />, {
+			registry,
+			viewContext: VIEW_CONTEXT_AD_BLOCKING_RECOVERY,
+		} );
+
+		container = renderResult.container;
+		getByRole = renderResult.getByRole;
```

Comment:
See above

Reply:
This becomes too much repetitive for the later steps without this pattern as we need a couple of act for navigating the the desired step.